### PR TITLE
tests: add tests for tld and domain transformations. 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #98

This PR adds tests for new transformations: tld and domain and tests code added in https://github.com/OISF/suricata/pull/4035

This update modifies the domain rule per updated domain transform logic.

### CODE REVIEW

File: tests/detect-domain-01/suricata.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+%YAML 1.1
```
Comment (victorjulien) [inline tests/detect-domain-01/suricata.yaml]: Do the tests not work if this is left out? A yaml should only be added to the tests if the default config is not sufficient.

File: tests/detect-tld-01/suricata.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+%YAML 1.1
```
Comment (victorjulien) [inline tests/detect-tld-01/suricata.yaml]: Same here, is the yaml really needed?

**Comment by jlucovsky** [comment]
Continued in #102

File: tests/detect-tld-01/suricata.yaml (hunk 2 above)
Comment (jlucovsky) [reply tests/detect-tld-01/suricata.yaml]: No; will remove.

File: tests/detect-domain-01/suricata.yaml (hunk 1 above)
Comment (jlucovsky) [reply tests/detect-domain-01/suricata.yaml]: No; will remove.

**Pull request closed without merging**